src/modules/rest/extjs/src/main/java/it/geosolutions/geostore/services/rest/impl/RESTExtJsServiceImpl.java

```diff
@@ -280,17 +296,26 @@ public String getResourcesByCategory(
         }
     }
 
+    private List<Resource> filterOutUnavailableResources(List<Resource> resources, User user) {
+        return resources.stream()
+                .filter(r -> isResourceAvailableForUser(r, user))
+                .collect(Collectors.toList());
+    }
+
     /*
      * (non-Javadoc)
      *
      * @see it.geosolutions.geostore.services.rest.RESTExtJsService#getResourcesList(javax.ws.rs.core.SecurityContext, java.lang.Integer,
-     * java.lang.Integer, boolean, boolean, it.geosolutions.geostore.services.dto.search.SearchFilter)
+     * java.lang.Integer, java.lang.String, java.lang.String, boolean, boolean, it.geosolutions.geostore.services.dto.search.SearchFilter)
      */
+
     @Override
     public ExtResourceList getExtResourcesList(
             SecurityContext sc,
             Integer start,
             Integer limit,
+            String sortBy,
+            String sortOrder,
```

Comment:
Similar to the approach mentioned earlier, a single request object could simplify the parameter list. This also future-proofs the code for additional flags in the future.

Reply:
Created a specific DTO sort object. Thanks.